recipes-omnect/initrdscripts/omnect-os-initramfs/factory-reset

```diff
@@ -112,6 +112,11 @@ factory_reset_finish() {
     # set return status
     set_bootloader_env_var ${FACTORY_RESET_STATUS_VAR} "${FACTORY_RESET_RET1}:${FACTORY_RESET_RET2}"
 
+    # if we had restore handling we have to clean unmount
+    if [ -n "${factory_reset_restore_list}" ]; then
```

Comment:
does that really work?
I see variable factory_reset_restore_list being set and used in factory_reset_run() but don't see it defined as global variable in that file.

Reply:
yes. it works reproducible. are not all non local variables, global? 
`factory_reset_finish()`  is called from within `factory_reset_run()` after setting `factory_reset_restore_list`. so it is in scope. do you want me to refactor it to use explicit local vars?

Reply:
no, it's ok, until just now I never saw the implication that using a non-local variable in a function does set it always globally! I was only aware that an assignment in a function will change an already defined global variable.
but it makes perfect sense that this works either way, of course.